src/Client/DurableStorage/FileStorage.php

```diff
@@ -2,13 +2,15 @@
 
 namespace Tuf\Client\DurableStorage;
 
+use Tuf\Metadata\StorageBase;
+
 /**
  * Defines a simple filesystem-based storage for fetched PHP-TUF metadata.
  *
  * Applications might want to provide an alternative implementation with
  * better performance and error handling.
  */
-class FileStorage implements \ArrayAccess
+class FileStorage extends StorageBase
```

Comment:
Does composer plugin extend this? If not do we want to take this chance to make this `final` since we are removing public methods and if anyone had extend it before we would break their code

Reply:
It does extend this class, but doesn't override any methods. It just adds helpful static factory methods. I don't mind making this final, but IMHO that should wait until the whole library is in a more final state.

Reply:
Maybe we could make a "Make all possible classes final" issue as beta blocker so we don't forget

Reply:
Opened #285 for that, along with a milestone for beta.